plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -324,19 +320,22 @@ func (l *windowsEventLog) getRecords(handles []EvtHandle) (records []*windowsEve
 
 		var publisherMetadataEvtHandle EvtHandle
 		if publisherMetadataEvtHandle, err = EvtOpenPublisherMetadata(0, publisher, nil, 0, 0); err != nil {
-			return nil, err
+			log.Printf("I! [windows_event_log] EvtOpenPublisherMetadata() err %v, publisher %v", err, newRecord.System.Provider.Name)
+			continue
```

Comment:
This is where the user's issue was coming from.
I didn't think these failures warranted an ERROR.
It is expected that some events in the event log cannot be rendered, which explains why the user had this issue in the first place.

Reply:
Windows event log publishers typically register before writing events to a specific event log (i.e. "application", "system", "security"). It is possible to write an event without doing so though.

PowerShell won't let you do it:
```
PS C:\Users\Administrator\go\src\amazon-cloudwatch-agent> Write-EventLog -logname "Application" -Source "AdamSource" -EventId 123 -EntryType "Error" -Message "hello world"
Write-EventLog : The source name "AdamSource" does not exist on computer "localhost".
At line:1 char:1
+ Write-EventLog -logname "Application" -Source "AdamSource" -EventId 1 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Write-EventLog], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteEventLogCommand
```

You must first call:
```
New-EventLog -LogName "Application" -Source "AdamSource"
```

You can do it with Python though:
```
"""
This is writing to the Windows application event log.
However the LOGSOURCE (applicationName) has not been created.
So the event will not display correctly in windows event viewer.
"""
import win32api
import win32con
import win32evtlog
import win32security
import win32evtlogutil

proc_handle = win32api.GetCurrentProcess()
proc_token = win32security.OpenProcessToken(proc_handle, win32con.TOKEN_READ)
my_sid = win32security.GetTokenInformation(proc_token, win32security.TokenUser)[0]

applicationName = "BarryPotter"
eventID = 321
category = 5    # Shell
myType = win32evtlog.EVENTLOG_ERROR_TYPE
descr = ["An error", "you know what it is"]
data = "Snape kills Dumbledore".encode("ascii")

win32evtlogutil.ReportEvent(applicationName, eventID, eventCategory=category,
    eventType=myType, strings=descr, data=data, sid=my_sid)
```

Note that even the windows event viewer will display the event, but the message is not displayed - which is expected, right?

So without digging into the how or why the Chrome install also causes this behavior, let's just leave it at - _it is possible_.

Reply:
MSDN:
https://docs.microsoft.com/en-us/windows/win32/eventlog/event-sources
```
An application can use the Application log without adding a new event source to the registry. If the application calls RegisterEventSource and passes a source name that cannot be found in the registry, the event-logging service uses the Application log by default. However, because there are no message files, the Event Viewer cannot map any event identifiers or event categories to a description string, and will display an error. For this reason, you should add a unique event source to the registry for your application and specify a message file.
```